typescript/demo/test-hf.ts

```diff
@@ -0,0 +1,74 @@
+import OpenAI from "openai";
+import * as path from "path";
+import { AIConfigRuntime } from "../lib/config";
+import { HuggingFaceTextGenerationModelParser } from "../lib/parsers/hf";
+import { Prompt } from "../types";
+
+async function run() {
+  const config = AIConfigRuntime.load(
+    path.join(
+      __dirname,
+      "../../python/tests/aiconfigs/basic_chatgpt_query_config.json"
+    )
+  );
+
+  // register HF MP
+  const hfMP = new HuggingFaceTextGenerationModelParser(
+    "mistralai/Mistral-7B-v0.1",
+    true
+  );
+  AIConfigRuntime.registerModelParser(hfMP, ["mistralai/Mistral-7B-v0.1"]);
```

Comment:
You shouldn't need to register `["mistralai/Mistral-7B-v0.1"])` again, right?

Reply:
In fact this is where using the `model_parsers` would be better

Reply:
Yes with the new schema update this isn't necessary! Will update on a new diff with changes in the comment below this as well, as this would entail a couple of small changes:
- Move construction of the HF Client to run to address api key
- Does not require  modelName; id will always return `HuggingFaceTextGeneration` parser 
- test harness doesn't need to instantiate a modelParser